### ISSUE TITLE
Need demo that opens/closes dialog, and accessibility checklist

### ISSUE THREAD

**Issue by tjsavage** [PolymerElements/paper-dialog-behavior#3]
If this is a demo of a dialog, we need the demo to show off the typical interaction: have a button that pops up the dialog, ideally not full-screen.

Things to check:
* Should have role=dialog, aria-modal=true/false appropriately
* Opening the dialog should place focus inside, or if there's nothing focusable, on the dialog element itself (tabindex=-1)
* When the dialog is open, other controls behind it on the page should be inert and all content outside the dialog should be aria-hidden
* Esc should close the dialog
* If the dialog has a title, use aria-labelledby to reference the title from the dialog element if possible. If impossible due to shadow boundaries, extract the title text and set the dialog's aria-label to that text.

**Comment by morethanreal** [comment]
As of 8ca5bcfd11a3e98998abe70cc02613a69be03c2c,

* `role="dialog"` is implemented
* `aria-modal="true"` is set if the dialog is `modal`, `aria-modal="false"` otherwise
* dialog has `tabindex=-1`, and it or an element with `autofocus` is focused when the dialog is opened
* esc closes the dialog, unless `no-cancel-on-esc-key` is set
* `aria-labelledby` points to the dialog title (a child `<h2>`) element if there is one

The only missing a11y request is to `aria-hidden` everything except the dialog, which is impractical for us.

**Comment by notwaldorf** [comment]
This has been fixed, minus the `aria-hidden` bit which is not possible
